src/util/paging.js

```diff
@@ -0,0 +1,46 @@
+const assert = require('assert');
+
+const objectEncode = obj => Buffer.from(JSON.stringify(obj)).toString('base64');
+
+const objectDecode = base64 => JSON.parse(Buffer.from(base64, 'base64').toString('utf8'));
+
+module.exports.fromCursor = (cursor) => {
+  let cursorPayload = {};
+  if (cursor !== null) {
+    try {
+      cursorPayload = objectDecode(cursor);
+    } catch (err) {
+      assert(err.name === 'SyntaxError' && err.message.startsWith('Unexpected token'), err);
```

Comment:
Is there is a problem with the returned/transmitted token the client would page indefinite...

I would just throw a new error "InvalidPageCursor" or something like that and then handle it outside the library

Reply:
Oh damn, that's a good point. I'll do exactly that